refactor(core): replace usages of removeChild

crisbeto opened the pull request:
These changes replace most usages of `removeChild` with `remove`. The latter has the advantage of not having to look up the `parentNode` and ensure that the child being removed actually belongs to the specific parent.

The refactor should be fairly safe since all the browsers we cover support `remove`. [Something similar was done in Components](https://github.com/angular/components/pull/23592) some time ago and there haven't been any bug reports as a result.

crisbeto commented:
[Mostly passing TGP](http://test/OCL:657881986:BASE:657935569:1722420408074:28495ae7), aside from one unrelated failure and one that I'll clean up separately.

josephperrott reviewed (approved):
LGTM

Reviewed-for: fw-security

alxhub reviewed (approved):
Reviewed-for: global-approvers

crisbeto commented:
Will run another TGP just in case.

crisbeto committed:
refactor(core): replace usages of removeChild

These changes replace most usages of `removeChild` with `remove`. The latter has the advantage of not having to look up the `parentNode` and ensure that the child being removed actually belongs to the specific parent.

The refactor should be fairly safe since all the browsers we cover support `remove`. [Something similar was done in Components](https://github.com/angular/components/pull/23592) some time ago and there haven't been any bug reports as a result.

crisbeto commented:
[Passing TGP from today](http://test/OCL:657881986:BASE:660394986:1723043496444:77456992)

thePunderWoman commented:
This PR was merged into the repository by commit 513a4fe05e26a99935f31d2f99a13dc373b612f0.

The changes were merged into the following branches: main, 18.1.x
